In [4]:
import pandas as pd
data = pd.read_csv('Expanded_data_with_more_features.csv')
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30641 entries, 0 to 30640
Data columns (total 15 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Unnamed: 0           30641 non-null  int64  
 1   Gender               30641 non-null  object 
 2   EthnicGroup          28801 non-null  object 
 3   ParentEduc           28796 non-null  object 
 4   LunchType            30641 non-null  object 
 5   TestPrep             28811 non-null  object 
 6   ParentMaritalStatus  29451 non-null  object 
 7   PracticeSport        30010 non-null  object 
 8   IsFirstChild         29737 non-null  object 
 9   NrSiblings           29069 non-null  float64
 10  TransportMeans       27507 non-null  object 
 11  WklyStudyHours       29686 non-null  object 
 12  MathScore            30641 non-null  int64  
 13  ReadingScore         30641 non-null  int64  
 14  WritingScore         30641 non-null  int64  
dtypes: float64(1), int64(4), object(10)


 The aim of this project is to study students grades and classify them under specific categories such as Excellent, Good, and Satisfactory,
while discovering the factors that affect them using data mining techniques. Data mining algorithms will be applied to extract patterns 
and relationships between students' grades and various factors, aiming to identify the main influences on academic performance.

 The source of the dataset https://www.kaggle.com/datasets/desalegngeb/students-exam-scores

In [5]:
rows, columns = data.shape
print(f"number of rows : {rows}, number of columns : {columns}")
print("names of attributes :\n", data.columns)
print("type of attributes :\n", data.dtypes)

number of rows : 30641, number of columns : 15
names of attributes :
 Index(['Unnamed: 0', 'Gender', 'EthnicGroup', 'ParentEduc', 'LunchType',
       'TestPrep', 'ParentMaritalStatus', 'PracticeSport', 'IsFirstChild',
       'NrSiblings', 'TransportMeans', 'WklyStudyHours', 'MathScore',
       'ReadingScore', 'WritingScore'],
      dtype='object')
type of attributes :
 Unnamed: 0               int64
Gender                  object
EthnicGroup             object
ParentEduc              object
LunchType               object
TestPrep                object
ParentMaritalStatus     object
PracticeSport           object
IsFirstChild            object
NrSiblings             float64
TransportMeans          object
WklyStudyHours          object
MathScore                int64
ReadingScore             int64
WritingScore             int64
dtype: object


## phase1


General information

number of columns: 15.

number of rows: 30641.

types of attributes:
ID - integer - float

class labal: Excellent, Good, Acceptable, and Fail.
